pkg/harvester/components/settings/upgrade-config.vue

```diff
@@ -0,0 +1,177 @@
+<script>
+import CreateEditView from '@shell/mixins/create-edit-view';
+import { LabeledInput } from '@components/Form/LabeledInput';
+import LabeledSelect from '@shell/components/form/LabeledSelect';
+import { RadioGroup } from '@components/Form/Radio';
+
+const DEFAULT_VALUE = {
+  imagePreloadOption: {
+    strategy: {
+      type:        'sequential',
+      concurrency: 0
+    }
+  },
+  restoreVM: false
+};
+
+export default {
+  name: 'HarvesterUpgradeConfig',
+
+  components: {
+    LabeledInput,
+    LabeledSelect,
+    RadioGroup
+  },
+  mixins: [CreateEditView],
+
+  data() {
+    let parseDefaultValue = {};
+
+    try {
+      parseDefaultValue = this.value.value ? JSON.parse(this.value.value) : { ...DEFAULT_VALUE };
+    } catch (error) {
+      parseDefaultValue = { ...DEFAULT_VALUE };
+    }
+
+    // Defensive: ensure all fields exist with defaults
+    if (!parseDefaultValue.imagePreloadOption) {
+      parseDefaultValue.imagePreloadOption = { strategy: { type: 'sequential', concurrency: 0 } };
+    }
+    if (!parseDefaultValue.imagePreloadOption.strategy) {
+      parseDefaultValue.imagePreloadOption.strategy = { type: 'sequential', concurrency: 0 };
+    }
+    if (!parseDefaultValue.imagePreloadOption.strategy.type) {
+      parseDefaultValue.imagePreloadOption.strategy.type = 'sequential';
+    }
+    if (typeof parseDefaultValue.imagePreloadOption.strategy.concurrency !== 'number') {
+      parseDefaultValue.imagePreloadOption.strategy.concurrency = 0;
+    }
+    if (typeof parseDefaultValue.restoreVM !== 'boolean') {
+      parseDefaultValue.restoreVM = false;
+    }
+
+    return {
+      parseDefaultValue,
+      errors: []
+    };
+  },
+
+  computed: {
+    strategyOptions() {
+      return [
+        { value: 'sequential', label: 'sequential' },
+        { value: 'skip', label: 'skip' },
+        { value: 'parallel', label: 'parallel' }
+      ];
+    }
+  },
+
+  created() {
+    this.update();
+  },
+
+  methods: {
+    update() {
+      try {
+        // Clone to avoid mutating the form state
+        const valueToSave = JSON.parse(JSON.stringify(this.parseDefaultValue));
+
+        if (valueToSave.imagePreloadOption && valueToSave.imagePreloadOption.strategy) {
+          if (valueToSave.imagePreloadOption.strategy.type !== 'parallel') {
+            delete valueToSave.imagePreloadOption.strategy.concurrency;
+          }
+        }
+        this.value['value'] = JSON.stringify(valueToSave, null, 2);
+        this.errors = [];
+      } catch (e) {
+        this.errors = ['Invalid JSON'];
+      }
+    }
```

Comment:
It should not display the `Modified` when the user is using the default value.
![Screenshot 2025-06-25 at 12 27 24 PM (2)](https://github.com/user-attachments/assets/1da64979-8ce1-4233-86e9-c937e12ffef4)